src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -81,11 +81,10 @@ object DepositHandler {
 
   def finalizeDeposit(mimeType: String)(implicit settings: Settings, id: String): Try[Unit] = {
     log.info(s"[$id] Finalizing deposit")
-    implicit val bagStoreBase = BagStoreBase(settings.bagStoreBaseDir, settings.bagStoreBaseUrl)
+    implicit val bagStoreSettings = settings.bagStoreSettings
```

Comment:
Don't think you need this implicit here. Just pass on the `settings` object to the subsequent calls and get you `Option[BagStoreSettings]` there.